### ISSUE TITLE
Edit history does not expand if URL is updated in Firefox

### ISSUE THREAD

**Issue by trichoplax** [codidact/qpixel#1056]
**Describe the bug**
Following #1030 there is a subtle problem when pasting a copied edit history link into the same tab.

Since it's a minor problem I'm raising it here as a separate issue, to avoid holding up approval of the mentioned pull request since it's a definite improvement.

**To Reproduce**
Steps to reproduce the behavior ~(will only work after #1030 is released, or in development)~:
1. Copy a link to edit number 1 in the edit history of a post
1. Paste this into a new tab
1. Copy a link to edit number 2 in the edit history of the same post
1. Paste this into the same tab as previously
1. The page still shows edit number 1 expanded, rather than edit number 2 (at least in Firefox)

In Chrome the correct edit number is expanded, but the others are not collapsed if they were already expanded before updating the URL.

**Expected behavior**
Requirement: The edit corresponding to the number in the URL should be automatically expanded when the URL is changed.

Optional behaviour (to be discussed): Any other open edits should be automatically collapsed, when the URL is changed.

**Desktop**
 - OS: Fedora 38 (no reason to think this will be different for other operating systems)
 - Browser: Firefox 113.0.1 (64 bit)

**Comment by ArtOfCode-** [comment]
@trichoplax If you have a dev environment, can you test: at https://github.com/codidact/qpixel/blob/develop/app/assets/javascripts/post_histories.js#L2, change `prop` to `attr` and see if that makes a difference?

**Comment by trichoplax** [comment]
@ArtOfCode- I just tried and unfortunately the behaviour seems the same.

Firefox: pasting in a new URL does not result in any change. Only on a page refresh does the correct edit expand and the others collapse.

Chromium: pasting in a new URL results in the correct edit expanding, but none of the others collapsing (they stay in whichever state they were already in). Only on a page refresh do the other edits collapse.

**Comment by ArtOfCode-** [comment]
I wonder if this is actually a browser thing - the expander code runs once on doc-ready, so depends on when the browser fires it. For instance, if Chrome does reload the page when you only change the anchor but Firefox doesn't, that would result in this behaviour. Don't know if that's how it actually works, but that could be a cause.

**Comment by trichoplax** [comment]
From the previous pull request notes, it sounds like even on a first load in a fresh tab, Chrome automatically expands when necessary to show the element with the specified id. So the expander code is only necessary for Firefox even in a fresh tab.

I guess this might just need the expander code to detect a change of URL rather than doc-ready, then it might help both Chrome and Firefox.

**Comment by MoshiKoi** [comment]
That is correct; Chromium browsers will automatically expand `details` elements if the jump anchor is inside of it, even without the script.

**Comment by trichoplax** [comment]
My proposed fix means Chromium will no longer automatically expand the relevant edit in the edit history, but since the JavaScript is handling this for both browsers hopefully this is acceptable. This way there is only one instance of each id.
